src/expr/core/src/expr/expr_udf.rs

```diff
@@ -126,7 +128,12 @@ impl UserDefinedFunction {
             UdfImpl::JavaScript(runtime) => runtime.call(&self.identifier, &input)?,
             UdfImpl::External(client) => {
                 let disable_retry_count = self.disable_retry_count.load(Ordering::Relaxed);
-                let result = if disable_retry_count != 0 {
+                let result = if self.always_retry_on_network_error {
```

Comment:
Should we put all `disable_retry_count` related stuff in `else` branch?

Reply:
Hmm no difference to me? Seems to add more nesting.

Reply:
I think mixing them together can lead to confusion. 

If nesting doesn't look good, we can add a sth like `call_with_ disable_retry_count`...

Reply:
Ah I didn't get what you meant originally. Now I do. Updated it.